definitions/variable/technology/electricity-operation.yaml

```diff
@@ -12,360 +12,71 @@
-    description: Active power of Biomass power plants
-    unit: MWh
-- Active Power|Electricity|Biomass|Traditional:
-    description: Active power of  Traditional Biomass power plants biomass from local
-      and pre-existing resources mostly waste :domestic and agricultural waste, forest
-      residues.
-    unit: MWh
-- Active Power|Electricity|Biomass|New and imported:
-    description: Active power of New+Imported biomass power plants biomass from dedicated
-      energy crops and biomass imports.
-    unit: MWh
-- Active Power|Electricity|Nuclear:
-    description: Active power of nuclear power plants
-    unit: MWh
-- Active Power|Electricity|Gas:
-    description: Active power of Gas power plants
-    unit: MWh
-- Active Power|Electricity|Gas|OCGT:
-    description: Active power of Open Cycle Gas Turbine power plants
-    unit: MWh
-- Active Power|Electricity|Gas|CCGT:
-    description: Active power of Combined Cycle Gas Turbine power plants
-    unit: MWh
-- Active Power|Electricity|Coal:
-    description: Active power of coal power plants
-    unit: MWh
-- Active Power|Electricity|Lignite:
-    description: Active power of Lignite power plants
-    unit: MWh
-- Active Power|Electricity|Oil:
-    description: Active power of oil power plants
-    unit: MWh
-- Active Power|Electricity|Geothermal:
-    description: Active power of Geothermal power plants
-    unit: MWh
-- Active Power|Electricity|Hydro|Reservoir:
-    description: Active power of "simple Hydro reservoir " units
-    unit: MWh
-- Active Power|Electricity|Hydro|Pumped Storage:
-    description: Active power of "Pumped Storage " units
-    unit: MWh
-- Active Power|Electricity|Hydro|Run of River:
-    description: Active power of "Run of River" units
-    unit: MWh
-- Active Power|Electricity|Hydro|Ocean:
-    description: Active power of Ocean power plants
-    unit: MWh
-- Active Power|Electricity|Solar:
-    description: Active power of Photovoltaic power plants
-    unit: MWh
-- Active Power|Electricity|Wind|OnShore:
-    description: Active power of  Wind Onshore power plants
-    unit: MWh
-- Active Power|Electricity|Wind|OffShore:
-    description: Active power of  Wind OffShore power plants
-    unit: MWh
+- Active Power|Electricity|<Fuel>:
+    description: Active power of <this fuel> power plants
+    unit:
+    - GWh
+
 - Active Power|Electricity|Load Curtailment:
     description: Energy curtailed - from the system point of view is equivalent to
       generation of Active Power
-    unit: MWh
+    unit:
+    - GWh
+
 - Active Power|Electricity|Load Shifting:
     description: Energy shifted - from the system point of view is equivalent to generation
       of Active Power
-    unit: MWh
+    unit:
+    - GWh
+
 - Active Power|Electricity|Electricity Storage:
     description: Active Power of energy storage systems, excl. hydro storage
-    unit: MWh
+    unit:
+    - GWh
+
+- Energy Consumption|Electricity|Energy Storage System:
```

Comment:
Sorry for being a pain here, but I think that this is ambiguous - do you mean the total withdrawal of energy form the grid, and if yes, why is this not just a negative value of *Active Power|Electricity|Electricity Storage*? Or is this the actual consumption for operating the storage facility, ie the losses = difference between withdrawal and injection?

Reply:
No worries, I think it is a perfect moment to clarify it.
Commonly, there are two options two to represent the production & consumption of any device that has a storage if I am not wrong:
1.  Two variables to represent the production & consumption separately are also considered in the mathematical formulation of the model.
2. One variable to represent the total withdrawal means `production + losses = consumption`; however, in some cases, the mathematical formulation internally considers production, losses, and consumption as variables.

Then, as we consider the production, losses, and consumption as variables within our models, the natural way is the `option 1` that also as the advantage to show the losses in an explicitly way instead of the `option 2` that the losses are implicit and need to be recovered.

I would prefer option 1, even if it means including an additional variable.
In addition, I'll consult about it to the responsibles in the case study.

Reply:
Not that I am expert on this, but I would vote for option 1 here. (I will also need to use this in our CS)

Reply:
@sandrinecharousset, could you please also comment on this question?

Beyond that, even if we have two variables for injection into the grid and withdrawal from the grid of an electricity storage facility, I think that "Consumption" is a misleading word for it.

Reply:
Argh, I had written a long message but then everything disapeared.......
In summary my feeling is that Active power is enough, but I am open to other views.
For describing an energy storage system (in power-plant.yma), I need Maximum Charge, Maximum Discharge, Maximum Storage, Minimum Storage, Charging efficiency and discharging efficiency, which are the 'physical' characteristcs of the storage;
Then I also need to know the "forced" interactions with the rest of the world, which in my model is summarised in a variable Inflows (a time series), which is positive if energy is being added to the system, negative if it means a demand served by the ESS (not through the grid)
Regarding the operation variables, ActivePower for me is enough if we consider no spillage; But if we consider spillage is possible then we need to have also a Spillage variable. I understand the possibility of having 'consumption' and 'production' but I find this less understandable, and also it would double the size of the data considering that at each timestep you would have both consumption and production instead of only ActivePower...... which in my case (eg in CS1 case where we model appliances with flexibility as storage systems) would mean lots of additionnal rows... (but this is a minor consideration)

Reply:
I suggest to remove the Spillage variable and stick to Active power being negative

Reply:
I suggest to only keep 'Active power' which when positive means the storage system is discharging energy to the grid, and when negative is charging energy from the grid